src/DotNetLightning.Core/Utils/SeqConsumer.fs

```diff
@@ -0,0 +1,58 @@
+namespace DotNetLightning.Utils
+
+type SeqConsumer<'S, 'T> = {
+    Consume: seq<'S> -> Option<seq<'S> * 'T>
+}
+
+type SeqConsumerBuilder<'S>() =
+    member __.Bind<'T0, 'T1>(seqConsumer0: SeqConsumer<'S, 'T0>, func: 'T0 -> SeqConsumer<'S, 'T1>)
```

Comment:
@canndrew I'm not a fan of generic types being named this way, you can put more verbose names, like we do in some places in geewallet e.g. https://gitlab.com/nblockchain/geewallet/-/blob/master/src/GWallet.Backend/FaultTolerantParallelClient.fs#L127